src/VisualStudio/Xaml/Impl/Implementation/XamlOleCommandTarget.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.VisualStudio.LanguageServices.Xaml
+{
+    /// <summary>
+    /// This command target routes commands in .xaml files.
+    /// </summary>
+    internal sealed class XamlOleCommandTarget : AbstractOleCommandTarget
+    {
+        internal XamlOleCommandTarget(
+            IWpfTextView wpfTextView,
+            ICommandHandlerServiceFactory commandHandlerServiceFactory,
+            IVsEditorAdaptersFactoryService editorAdaptersFactory,
+            IServiceProvider serviceProvider)
+            : base(wpfTextView, commandHandlerServiceFactory, editorAdaptersFactory, serviceProvider)
+        {
+            wpfTextView.Closed += OnTextViewClosed;
+            wpfTextView.BufferGraph.GraphBufferContentTypeChanged += OnGraphBuffersChanged;
+            wpfTextView.BufferGraph.GraphBuffersChanged += OnGraphBuffersChanged;
```

Comment:
The very next thing that happens after the ctor is a call to AttachToVsTextView(), which besides other things does the exact same thing as RefreshCommandFilters(), so it would essentially call it twice. I have a feeling that the other derived types call this unnecessarily.